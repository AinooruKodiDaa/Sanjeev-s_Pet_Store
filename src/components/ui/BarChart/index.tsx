import React, { useEffect, useRef } from 'react';
import * as echarts from 'echarts';
import { Typography } from '../Typography';

interface DataItem {
  name: string;
  value: number;
}

interface BarChartProps {
  data: DataItem[];
  colors: { [key: string]: string };
}

const BarChart: React.FC<BarChartProps> = ({ data, colors }) => {
  const chartRef = useRef<HTMLDivElement>(null);

  useEffect(() => {
    if (!chartRef.current) return;

    const chart = echarts.init(chartRef.current);

    const options: echarts.EChartsOption = {
      xAxis: {
        type: 'category',
        data: data.map((item: DataItem) => item.name)
      },
      yAxis: {
        type: 'value',
       
    },
      series: [{
        data: data.map((item: DataItem) => ({
          value: item.value,
          itemStyle: {
            color: colors[item.name] || '#2196f3' // Default color if specific color not provided
          }
        })),
        type: 'bar'
      }]
    };

    chart.setOption(options);

    return () => {
      chart.dispose(); // Dispose chart instance on unmounting
    };
  }, [data, colors]);

  return <div ref={chartRef} style={{ width: '100%', height: '100%' }} />;
};

export default BarChart;
