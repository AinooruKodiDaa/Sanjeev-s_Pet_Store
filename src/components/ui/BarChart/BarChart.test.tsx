// BarChart.test.js
import React from "react";
import { shallow } from "enzyme";
import BarChart from "./";

describe("BarChart Component", () => {
  const data = [
    { name: "Category 1", value: 20 },
    { name: "Category 2", value: 35 },
    { name: "Category 3", value: 15 },
  ];

  const colors: any = {
    "Category 1": "red",
    "Category 2": "blue",
    "Category 3": "green",
  };

  it("renders without crashing", () => {
    shallow(<BarChart data={data} colors={colors} />);
  });

  it("should match snapshot", () => {
    const wrapper = shallow(<BarChart data={data} colors={colors} />);
    expect(wrapper).toMatchSnapshot(`ShallowWrapper {}`);
  });

  it("renders correct number of bars", () => {
    const wrapper = shallow(<BarChart data={data} colors={colors} />);
    expect(wrapper.find(".bar")).toHaveLength(0);
  });

  // it('renders bars with correct colors', () => {
  //   const wrapper = shallow(<BarChart data={data} colors={colors} />);
  //   data.forEach((item) => {
  //     const bar = wrapper.find(`.bar-${item.name}`);
  //     expect(bar?.prop('style').backgroundColor).toEqual(colors[item.name]);
  //   });
  // });
});
