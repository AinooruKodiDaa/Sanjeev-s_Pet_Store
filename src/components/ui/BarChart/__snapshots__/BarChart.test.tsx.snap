// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarChart Component  1`] = `ShallowWrapper {}`;

exports[`BarChart Component should match snapshot: ShallowWrapper {} 1`] = `ShallowWrapper {}`;

exports[`BarChart Component to match snapshot 1`] = `ShallowWrapper {}`;
